jms/components/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/listener/JMSServerConnector.java

```diff
@@ -0,0 +1,188 @@
+                logger.error("Provided value for max retry count is invalid, using the default max retry count "
+                        + maxRetryCount);
+            }
+        }
+
+    }
+
+
+    @Override
+    public boolean bind() {
+        try {
+            createDestinationListener();
+            return true;
+        } catch (Exception e) {
+            try {
+                if (jmsConnectionFactory == null) {
+                    throw new RuntimeException("Cannot create the jms connection factory. please check the connection"
+                            + " properties and re-deploy the jms service");
+                }
+                closeAll();
+                JMSConnectionRetryHandler jmsConnectionRetryHandler = new JMSConnectionRetryHandler(this,
+                        retryInterval, maxRetryCount);
+                jmsConnectionRetryHandler.run();
+            } catch (JMSServerConnectorException e1) {
+                throw new RuntimeException(e1.getMessage(), e1);
+            }
+        }
+        return false;
+    }
+
+    @SuppressFBWarnings({"REC_CATCH_EXCEPTION"})
+    void createDestinationListener() throws JMSServerConnectorException {
+        try {
+            jmsConnectionFactory = new JMSConnectionFactory(properties);
+            connection = jmsConnectionFactory.createConnection();
+            jmsConnectionFactory.start(connection);
+            session = jmsConnectionFactory.createSession(connection);
+            destination = jmsConnectionFactory.getDestination(session);
+            messageConsumer = jmsConnectionFactory.createMessageConsumer(session, destination);
+            messageConsumer.setMessageListener(
+                    new JMSMessageListener(carbonMessageProcessor, id, session.getAcknowledgeMode(), session));
+        } catch (Exception e) {
+            logger.error("Error while creating the connection from connection factory. " + e.getMessage());
+            throw new JMSServerConnectorException("Error while creating the connection from connection factory", e);
+        }
+
+    }
+
+    void closeAll() {
+        try {
+            jmsConnectionFactory.closeConnection(connection);
+            jmsConnectionFactory.closeSession(session);
+            jmsConnectionFactory.closeMessageConsumer(messageConsumer);
+        } catch (JMSServerConnectorException e) {
+            logger.error("Error while closing the connection, session and message consumer ", e);
+        }
+    }
+
+    @Override
+    public boolean unbind() {
+        return false;
+    }
+
+    public JMSServerConnector() {
+        super("jms");
+    }
+
+    @Override
+    public void setMessageProcessor(CarbonMessageProcessor carbonMessageProcessor) {
+        this.carbonMessageProcessor = carbonMessageProcessor;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected void start() {
+        // Not needed for JMS Transport
+    }
+
+    @Override
+    public void stop() {
```

Comment:
Shouln't this be moved to unbind since we need to stop everything when the server is undeployed?